src/GraphQL/Types/Composite/AutoRegisteringHelper.cs

```diff
@@ -69,5 +81,40 @@ private static string GetPropertyName<TSourceType>(Expression<Func<TSourceType,
 
             throw new NotSupportedException($"Unsupported type of expression: {expression.GetType().Name}");
         }
+
+        /// <summary>
+        /// Constructs a lambda expression for a field resolver to return the specified query argument
+        /// from the resolve context.
+        /// </summary>
+        internal static LambdaExpression GetParameterExpression(Type parameterType, QueryArgument queryArgument)
+        {
+            //construct a call to ResolveFieldContextExtensions.GetArgument, passing in the appropriate default value
+            var getArgumentMethodTyped = _getArgumentMethod.MakeGenericMethod(parameterType);
+            var resolveFieldContextParameter = Expression.Parameter(typeof(IResolveFieldContext), "context");
+            var queryArgumentExpression = Expression.Constant(queryArgument, typeof(QueryArgument));
+            //e.g. Func<IResolveFieldContext, int> = (context) => ResolveFieldContextExtensions.GetArgument<int>(context, queryArgument.Name, queryArgument.DefaultValue);
+            var expr = Expression.Call(getArgumentMethodTyped, resolveFieldContextParameter, queryArgumentExpression);
+
+            return Expression.Lambda(
+                expr,
+                resolveFieldContextParameter);
+        }
+
+        private static readonly MethodInfo _getArgumentMethod = typeof(AutoRegisteringHelper).GetMethod(nameof(GetArgumentInternal), BindingFlags.NonPublic | BindingFlags.Static)!;
+        /// <summary>
+        /// Returns the value for the specified query argument, or the default value of the query argument
+        /// if a value was not specified in the request.
+        /// <br/><br/>
+        /// Unlike <see cref="ResolveFieldContextExtensions.GetArgument{TType}(IResolveFieldContext, string, TType)"/>,
+        /// the default value is not returned if <see langword="null"/> was supplied.
```

Comment:
> if \<see langword="null"/> was supplied

supplied where?

Reply:
As a literal or variable. 

I feel that the getargument behavior is wrong. Something we should consider changing. But certainly the argument support here doesn’t work properly with the normal GetArgument behavior.

Reply:
> As a literal or variable.

Ah. I look for a null value in some method argument. Worth to add _was supplied in request as a literal or variable_.

Reply:
It's already been updated.  Suggest a change if you like.

Reply:
fine